### PR TITLE
Feat(eos_validate_state): Added the validation for AVT path for static peers and role of device

### LINKED ISSUE #4188: Feat(eos_validate_state): Add validation for AVT path state and role
### Enhancement summary

Add the validation of static AVT path state and role.

### Which component of AVD is impacted

eos_validate_state

### Use case example

Since we have AVT profile in AVD and static peers in router path selection so we can add the validation of AVT path state and role of device.

### Describe the solution you would like

Integrate the ANTA tests VerifyAVTSpecificPath and VerifyAVTRole into AVD

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Added the validation for AVT path for static peers and role of device.

## Related Issue(s)

Fixes #4188

## Component(s) name

`arista.avd.eos_validate_state`

## Proposed changes
Added the testcase for to verify AVT path and role from ANTA.

## How to test
Run the validation playbook on lab or run molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            {
+                peers_data["router_ip"]
+                for group_idx, path_group in enumerate(path_groups)
+                if self.validate_data(data=path_group, data_path=f"router_path_selection.path_groups.[{group_idx}]", required_keys="static_peers")
+                for peer_idx, peers_data in enumerate(path_group["static_peers"])
+                if self.validate_data(
+                    data=peers_data, data_path=f"router_path_selection.path_groups.[{group_idx}].static_peers.[{peer_idx}]", required_keys="router_ip"
+                )
+            }
+        )
```
Comment (laxmikantchintakindi) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py:56]:
```suggestion
static_peers_list = []
for group_idx, path_group in enumerate(path_groups):
 for peer_idx, peers_data in enumerate(path_group["static_peers"]):
  if self.validate_data((data=path_group, data_path=f"router_path_selection.path_groups.[{group_idx}]", required_keys="static_peers") and self.validate_data(
                    data=peers_data, data_path=f"router_path_selection.path_groups.[{group_idx}].static_peers.[{peer_idx}]", required_keys="router_ip"
                ):
   static_peers.append(peers_data["router_ip"]

static_peers = sorted(set(static_peers_list))
```

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        if not avt_profiles:
+            LOGGER.info("AVT profiles are not configured for any VRF. %s is skipped.", self.__class__.__name__)
+            return None
+
+        # Build a set of static peers
+        static_peers = sorted(
+            {
+                peers_data["router_ip"]
+                for group_idx, path_group in enumerate(path_groups)
+                if self.validate_data(data=path_group, data_path=f"router_path_selection.path_groups.[{group_idx}]", required_keys="static_peers")
```
Comment (laxmikantchintakindi) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py]: Is it possible to make this block easy using nested for loop.

Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py]: can we even maybe make it a helper function? like `_get_static_peers` or whaterver?

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            dict | None: ANTA test definition or None if the configuration is incomplete.
+        """
+        # Retrieve path groups from the structured configuration
+        path_groups = get(self.structured_config, "router_path_selection.path_groups")
+        if not path_groups:
+            LOGGER.info("Path groups are not configured to collect AVT peer information. %s is skipped.", self.__class__.__name__)
+            return None
+
+        # Retrieve AVT profiles from the structured configuration
+        avt_profiles = get(self.structured_config, "router_adaptive_virtual_topology.vrfs")
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py]: I am confused why vrf are called avt_profiles? under router_adaptive_virtual_topology you have a list of vrfs and under each vrf a list of profiles being listed right so not sure what we are picking up here

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                        }
+                    ],
+                    "result_overwrite": {
+                        "custom_field": f"AVT profile: {avt_profile['name']}, vrf: {vrf['name']}, "
+                        f"Destination IPv4 Address: {dst_address}, Nexthop VTEP: {dst_address}"
+                    },
+                }
+            }
+            for vrf in avt_profiles
+            for avt_profile in vrf["profiles"]
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py]: this is confusing - variable names must be changed

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md [hunk 5]
```diff
@@ -967,6 +967,8 @@ wan_virtual_topologies:
 | AvdTestBGP | VerifyBGPSpecificPeers | Validate the state of BGP Address Family sessions, including `Path-Selection` for AutoVPN, `Link-State` and `IPv4/IPv6 SR-TE` for CV Pathfinder. |
 | AvdTestIPSecurity | VerifySpecificIPSecConn | Validate the establishment of IP security connections for each static peer under the `router path-selection` section of the configuration. |
 | AvdTestStun | VerifyStunClient | Validate the presence of a STUN client translation for a given source IPv4 address and port. The list of expected translations for each device is built by searching local interfaces in each path-group. |
+| AvdTestAvtPath | VerifyAVTSpecificPath | Validate that the status is active and the type is direct for an Adaptive Virtual Topology (AVT) path in a specified VRF. |
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md]: need to explain that this is only checking the static peers (for now, lets see if we can be smarter later)

Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md]: change also the PR description and title to indicate this

**Review by gmuloc — DISMISSED** [review]
First round of review

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py (hunk 3 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py]: updated the variable names and right now i am collecting all vrfs then under that all profiles and then making all possible commands for each destination as in structure config there is no direct information about which destination is mapped with which vrf and avt profile

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md (hunk 5 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md]: updated

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py [hunk 6]
```diff
@@ -22,6 +22,31 @@ class AvdTestAvtPath(AvdTestBase):
...
+        """
+        static_peers = set()
+
+        for group_idx, path_group in enumerate(path_groups):
+            if self.validate_data(data=path_group, data_path=f"router_path_selection.path_groups.[{group_idx}]", required_keys="static_peers"):
+                for peer_idx, peers_data in enumerate(path_group["static_peers"]):
+                    if self.validate_data(
+                        data=peers_data, data_path=f"router_path_selection.path_groups.[{group_idx}].static_peers.[{peer_idx}]", required_keys="router_ip"
+                    ):
+                        static_peers.add(peers_data["router_ip"])
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py]:
```suggestion
            if not self.validate_data(data=path_group, data_path=f"router_path_selection.path_groups.[{group_idx}]", required_keys="static_peers"):
                continue
            for peer_idx, peers_data in enumerate(path_group["static_peers"]):
                if self.validate_data(
                    data=peers_data, data_path=f"router_path_selection.path_groups.[{group_idx}].static_peers.[{peer_idx}]", required_keys="router_ip"
                    ):
                    static_peers.add(peers_data["router_ip"])
```

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+                    "avt_paths": [
+                        {
+                            "avt_name": avt_profile["name"],
+                            "vrf": vrf["name"],
+                            "destination": dst_address,
+                            "next_hop": dst_address,
+                        }
+                    ],
+                    "result_overwrite": {
+                        "custom_field": f"AVT profile: {avt_profile['name']}, vrf: {vrf['name']}, "
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py]:
```suggestion
                        "custom_field": f"AVT profile: {avt_profile['name']}, VRF: {vrf['name']}, "
```

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md [hunk 8]
```diff
@@ -989,6 +989,8 @@ wan_virtual_topologies:
 | AvdTestBGP | VerifyBGPSpecificPeers | Validate the state of BGP Address Family sessions, including `Path-Selection` for AutoVPN, `Link-State` and `IPv4/IPv6 SR-TE` for CV Pathfinder. |
 | AvdTestIPSecurity | VerifySpecificIPSecConn | Validate the establishment of IP security connections for each static peer under the `router path-selection` section of the configuration. |
 | AvdTestStun | VerifyStunClient | Validate the presence of a STUN client translation for a given source IPv4 address and port. The list of expected translations for each device is built by searching local interfaces in each path-group. |
+| AvdTestAvtPath | VerifyAVTSpecificPath | Validate that the status is active and the type is direct for an Adaptive Virtual Topology (AVT) path in a specified VRF for the static peers. |
+| AvdTestAvtRole | VerifyAVTRole | Validate the Adaptive Virtual Topology (AVT) role of a device. |
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md:1046]: `eos_validate_state` doc needs to be updated as well. Thanks

**Review by carl-baillargeon — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+        # Construct the list of ANTA tests
+        anta_tests = [
+            {
+                "VerifyAVTRole": {
+                    "role": avt_role,
+                }
+            }
+        ]
+
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py]: This fails for transit region (words are reversed) so we need to translate here
```
    {
      "name": "site1-wan1",
      "test": "VerifyAVTRole",
      "categories": [
        "avt"
      ],
      "description": "Verifies the AVT role of a device.",
      "result": "failure",
      "messages": [
        "Expected AVT role as `transit region`, but found `region transit` instead."
      ]
    },
```

```
site1-wan1#show run se adap                                                                                                                                                                                               13:59:16 [32/12351]
router adaptive-virtual-topology
   topology role transit region
   region REGION1 id 1
site1-wan1#show adaptive-virtual-topology path  | json
{
    "routerId": "192.168.42.3",
    "role": "region transit",
    "regionName": "REGION1",
    "regionId": 1,
```

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+        Returns:
+            dict | None: ANTA test definition or None if the configuration is incomplete.
+        """
+        # Retrieve AVT role from the structured configuration
+        avt_role = get(self.structured_config, "router_adaptive_virtual_topology.topology_role")
+        if not avt_role:
+            LOGGER.info("AVT role is not configured. %s is skipped.", self.__class__.__name__)
+            return None
+
+        # Construct the list of ANTA tests
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py]:
```suggestion
        if not avt_role:
            LOGGER.info("AVT role is not configured. %s is skipped.", self.__class__.__name__)
            return None
       # Translating AVT role for transit region and zone due to JSON output
       if avt_role in ["transit region", "transit zone"]:
           avt_role = ' '.join(reversed(avt_role.split()))

        # Construct the list of ANTA tests
```

**Review by gmuloc — APPROVED** [review]

**Pull request merged**